From: Dustin Brown <dustinb@codeaurora.org>
Date: Tue, 13 Jun 2017 11:40:56 -0700
Subject: [PATCH] arm64: Export save_stack_trace_tsk()
Git-commit: e27c7fa015d61c8be6a2c32b2144aad2ae6ec975
Patch-mainline: v4.13
References: jsc#SLE-4214

The kernel watchdog is a great debugging tool for finding tasks that
consume a disproportionate amount of CPU time in contiguous chunks. One
can imagine building a similar watchdog for arbitrary driver threads
using save_stack_trace_tsk() and print_stack_trace(). However, this is
not viable for dynamically loaded driver modules on ARM platforms
because save_stack_trace_tsk() is not exported for those architectures.
Export save_stack_trace_tsk() for the ARM64 architecture to align with
x86 and support various debugging use cases such as arbitrary driver
thread watchdog timers.

Signed-off-by: Dustin Brown <dustinb@codeaurora.org>
Signed-off-by: Will Deacon <will.deacon@arm.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 arch/arm64/kernel/stacktrace.c |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm64/kernel/stacktrace.c
+++ b/arch/arm64/kernel/stacktrace.c
@@ -175,6 +175,7 @@ void save_stack_trace_tsk(struct task_st
 
 	put_task_stack(tsk);
 }
+EXPORT_SYMBOL_GPL(save_stack_trace_tsk);
 
 void save_stack_trace(struct stack_trace *trace)
 {
